web/src/components/IdeEditor/IdeEditor.js

```diff
@@ -1,51 +1,62 @@
-import { useContext, useEffect, Suspense, lazy } from 'react'
-import { isBrowser } from '@redwoodjs/prerender/browserUtils'
+import { useContext, Suspense, lazy } from 'react'
 import { IdeContext } from 'src/components/IdeToolbarNew'
 import { codeStorageKey } from 'src/helpers/hooks/useIdeState'
+import { requestRender } from 'src/helpers/hooks/useIdeState'
 const Editor = lazy(() => import('@monaco-editor/react'))
 
 const IdeEditor = () => {
-  const { state, dispatch } = useContext(IdeContext)
-
-  const scriptKey = 'encoded_script'
-  useEffect(() => {
-    // load code from hash if it's there
-    let hash
-    if (isBrowser) {
-      hash = window.location.hash
-    }
-    const [key, scriptBase64] = hash.slice(1).split('=')
-    if (key === scriptKey) {
-      const script = atob(scriptBase64)
-      dispatch({ type: 'updateCode', payload: script })
-    }
-  }, [])
-  useEffect(() => {
-    if (isBrowser) {
-      window.location.hash = ''
-    }
-  }, [state.code])
+  const { state, thunkDispatch } = useContext(IdeContext)
+  const ideTypeToLanguageMap = {
+    cadQuery: 'python',
+    openScad: 'cpp',
+  }
 
   function handleCodeChange(value, _event) {
-    dispatch({ type: 'updateCode', payload: value })
+    thunkDispatch({ type: 'updateCode', payload: value })
   }
   function handleSaveHotkey(event) {
```

Comment:
I had missed you added this, what a great addition! And for it to be agnostic of the CAD package; this is the kind of unifying stuff that's gonna make it so nice to switch between packages.